pyiso/caiso.py

```diff
@@ -424,42 +430,56 @@ def set_dt_index(self, df, date, hours, end_of_hour=True):
             offset = 0
 
         # create list of combined datetimes
-        dts = [datetime.combine(date, time(hour=(h+offset))) for h in hours]
+        dts = [datetime.combine(date, time(hour=(int(h)+offset))) for h in hours]
 
         # set list as index
         df.index = dts
 
         # utcify
         df.index = self.utcify_index(df.index)
 
-        # return
         return df
 
     def _generation_historical(self):
         # set up storage
         parsed_data = []
 
         # collect data
-        this_date = self.options['start_at'].date()
-        while this_date <= self.options['end_at'].date():
+        request_date = self.options['start_at'].astimezone(self.ca_tz).date()
+        local_end_at = self.options['end_at'].astimezone(self.ca_tz).date()
+        while request_date <= local_end_at:
             # set up request
-            url_file = this_date.strftime('%Y%m%d_DailyRenewablesWatch.txt')
+            url_file = request_date.strftime('%Y%m%d_DailyRenewablesWatch.txt')
```

Comment:
Lines 448-452 correctly localize date being used for Daily Renewables Watch report requests (which are local date URLs).